src/main/scala/org/scalacheck/ScalaCheckFramework.scala

```diff
@@ -81,20 +82,23 @@ private abstract class ScalaCheckRunner extends Runner {
 
     def execute(handler: EventHandler, loggers: Array[Logger],
       continuation: Array[Task] => Unit
-    ): Unit  = continuation(execute(handler,loggers))
+    ): Unit  = continuation(execute(handler,loggers.asInstanceOf[Array[Logger | Null]]).map(_.nn)) // java method array argument contents need to be nullable.
   }
 
   def rootTask(td: TaskDef) = new BaseTask(td) {
-    def execute(handler: EventHandler, loggers: Array[Logger]): Array[Task] =
+    def execute(handler: EventHandler | Null, loggers: Array[Logger | Null] | Null): Array[Task | Null] =
       props.map(_._1).toSet.toArray map { name =>
         checkPropTask(new TaskDef(td.fullyQualifiedName, td.fingerprint,
-          td.explicitlySpecified, Array(new TestSelector(name)))
+          td.explicitlySpecified, Array(new TestSelector(name)).asInstanceOf[Array[sbt.testing.Selector | Null]]) // java method array argument contents need to be nullable.
```

Comment:
Pass `Array[T]` as argument to java method that requires an array. Java arrays have type `Array[T | Null]`.

Reply:
This would be fixed by improvements to type inference for prototypes with unions.